report_xlsx/controllers/main.py

```diff
@@ -29,13 +31,18 @@ def report_routes(self, reportname, docids=None, converter=None, **data):
             xlsx = report.with_context(context).render_xlsx(
                 docids, data=data
             )[0]
+            report_name = report.report_file
+            if report.print_report_name and not len(docids) > 1:
```

Comment:
Here you should check that docids is set:
`if report.print_report_name and docids and not len(docids) > 1:`
Just realized that sometimes a report is launched without docids, and it will fail.
In that case is fair to not personalize the name because a main object is not given and the data property would contain the details for rendering the report.

Reply:
Thank you for noticing. I fixed per your request here and also in v12